src/components/chat/chat-state.ts

```diff
@@ -0,0 +1,354 @@
+import type IgcTextareaComponent from '../textarea/textarea.js';
+import type IgcChatComponent from './chat.js';
+import type { IgcChatComponentEventMap } from './chat.js';
+import type {
+  IgcChatOptions,
+  IgcMessage,
+  IgcMessageAttachment,
+} from './types.js';
+
+/**
+ * Internal state manager for the `<igc-chat>` component.
+ *
+ * Manages messages, input value, attachments, options, and event emissions.
+ */
+export class ChatState {
+  //#region Internal properties and state /** The host `<igc-chat>` component instance */
+  private readonly _host: IgcChatComponent;
+  /** Reference to the text area input component */
+  private _textArea: IgcTextareaComponent | null = null;
+  /** The current list of messages */
+  private _messages: IgcMessage[] = [];
+  /** Sorted copy of messages by timestamp */
+  private _sortedMessages: IgcMessage[] = [];
+  /** Chat options/configuration */
+  private _options?: IgcChatOptions;
+  /** List of current input attachments */
+  private _inputAttachments: IgcMessageAttachment[] = [];
+  /** Current input text */
+  private _inputValue = '';
+  /**
+   * Cache of accepted file types, organized into extensions, mimeTypes, and wildcardTypes
+   */
+  private _acceptedTypesCache: {
+    extensions: Set<string>;
+    mimeTypes: Set<string>;
+    wildcardTypes: Set<string>;
+  } | null = null;
+
+  private _suggestionsPosition: 'below-input' | 'below-messages' =
+    'below-messages';
+  //#endregion
+
+  //#region Public properties
+  /**
+   * Array of message IDs sorted by timestamp ascending.
+   */
+  public get sortedMessagesIds(): string[] {
+    return this._sortedMessages.map((m) => m.id);
+  }
+
+  /**
+   * Gets the list of chat messages.
+   */
+  public get messages(): IgcMessage[] {
+    return this._messages;
+  }
+
+  /**
+   * Sets the list of chat messages.
+   * Also sorts messages by timestamp and requests host update.
+   */
+  public set messages(value: IgcMessage[]) {
+    this._messages = value;
+    this._sortedMessages = value.slice().sort((a, b) => {
+      return a.timestamp.getTime() - b.timestamp.getTime();
+    });
```

Comment:
Not a fan, similar to the group by behavior in the list component.
Up to the developer to sort them, or we expose an API hook for the user to provide a sort comparator.